pysph/base/tests/test_opencl.py

```diff
@@ -4,8 +4,47 @@
 
 pytest.importorskip("pysph.base.opencl")
 
+import pyopencl as cl
+
 from pysph.base.utils import get_particle_array  # noqa: E402
-from pysph.base.opencl import DeviceHelper  # noqa: E402
+from pysph.base.opencl import DeviceHelper, DeviceArray  # noqa: E402
+
+
+class TestDeviceArray(TestCase):
+    def setUp(self):
+        self.ctx = cl.create_some_context(interactive=False)
+        self.queue = cl.CommandQueue(self.ctx)
+        self.dev_array = DeviceArray(self.queue, np.int32)
+
+    def tearDown(self):
+        self.dev_array = DeviceArray(self.queue, np.int32)
+
+    def test_reserve(self):
+        self.dev_array.reserve(64)
```

Comment:
I don't think the setUp method is doing much, instead just create the `DeviceArray` here and use that.